src/app/plugins/pubsub.plugin.ts

```diff
@@ -1,23 +1,116 @@
+import Constants from '../../common/constants';
 import { Plugin } from '../../common/plugin';
-import { IContainer, IMessage, ChannelType } from '../../common/types';
+import { ChannelType, IContainer, IHttpResponse, IMessage } from '../../common/types';
+import { RichEmbed } from 'discord.js';
 
 export class PubSubPlugin extends Plugin {
   public name: string = 'Pub Sub Plugin';
-  public description: string = 'Are publix chicken tender subs on sale?';
-  public usage: string = '';
-  public pluginAlias = [];
-  public permission: ChannelType = ChannelType.Bot;
+  public description: string =
+    'Get them prices for your subs, and a steamy picture of the subs you need in your life.';
+  public usage: string = 'pubSub <sub (optional)>';
+  public pluginAlias = ['sub', 'subs', 'sandwich', 'samwich', 'sarnie'];
+  public permission: ChannelType = ChannelType.Public;
+  public pluginChannelName: string = Constants.Channels.Public.Food;
+
+  private _API_URL: string = 'https://pubsub-api.dev/';
+  private _subs: string[] = [];
+  private _embedSubs = new RichEmbed();
 
   constructor(public container: IContainer) {
     super();
+    this.updateList();
+  }
+
+  private updateList() {
+    this.container.httpService
+      .get(`${this._API_URL}allsubs/`)
+      .then((response: IHttpResponse) => {
+        const subs = response.data;
+
+        this._subs = subs.map((subData: { name: string }) => {
+          return subData.name
```

Comment:
this is unneeded complexity with how you are serializing each entry and reflects later on in the code. its fine to keep it as `-` and replace user input of spaces to `-`